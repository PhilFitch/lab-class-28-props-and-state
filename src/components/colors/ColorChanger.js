import React from 'react';
import PropTypes from 'prop-types';

const ColorChanger = ({ backgroundColor }) => (
  <div style={{ 
    height: '200px', 
    width: '200px', 
    backgroundColor
  }}></div>
);

ColorChanger.propTypes = {
  backgroundColor: PropTypes.string.isRequired
};

export default ColorChanger;
