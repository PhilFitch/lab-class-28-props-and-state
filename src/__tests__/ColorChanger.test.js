import React from 'react';
import { shallow } from 'enzyme';
import ColorChanger from '../components/colors/ColorChanger';

describe('ColorChanger component', () => {
  it('renders ColorChanger', () => {
    const wrapper = shallow(<ColorChanger backgroundColor='blue' />);
    expect(wrapper).toMatchSnapshot();
  });
});
