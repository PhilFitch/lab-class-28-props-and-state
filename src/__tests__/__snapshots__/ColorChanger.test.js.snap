// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorChanger component renders ColorChanger 1`] = `
<div
  style={
    Object {
      "backgroundColor": "blue",
      "height": "200px",
      "width": "200px",
    }
  }
/>
`;
